src/liballoc/boxed.rs

```diff
@@ -1070,3 +1071,12 @@ impl<F: ?Sized + Future + Unpin> Future for Box<F> {
         F::poll(Pin::new(&mut *self), cx)
     }
 }
+
+#[unstable(feature = "box_str_from_iter", issue = "0")]
+impl FromIterator<char> for Box<str> {
+    fn from_iter<I: IntoIterator<Item = char>>(iter: I) -> Box<str> {
+        let mut buf = String::new();
+        buf.extend(iter);
+        buf.into_boxed_str()
```

Comment:
Can the string creation reuse `String::from_iter`?

```rust
String::from_iter(iter).into_boxed_str()
````

Reply:
You're right, thanks!